EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp

```diff
@@ -1418,6 +1422,19 @@ void Road::AddElevation(Elevation *elevation)
 	elevation_profile_.push_back((Elevation*)elevation);
 }
 
+void Road::AddSuperElevation(Elevation *super_elevation)
+{
+	// Adjust last super elevation length
+	if (super_elevation_profile_.size() > 0)
+	{
+		Elevation *e_previous = super_elevation_profile_.back();
+		e_previous->SetLength(super_elevation->GetS() - e_previous->GetS());
+	}
+	super_elevation->SetLength(length_ - super_elevation->GetS());
+
```

Comment:
I'm thinking what happens if first record is at s > 0, e.g. s = 100. I guess the range 0 < s < 100 will be undefined. A potential fall back could be to add/insert a dummy superelevation record with a=b=c=d=0 before first user added entry.
But on the other hand, maybe it's better that the user is made aware of the "bug" and add an entry from s=0...